### ISSUE TITLE
Push: Some non-empty files uploaded as zero bytes.

### ISSUE THREAD

**Issue by bopbo** [odeke-em/drive#365]
Hello,

I installed drive on CentOS 7 and it's working but when i push large files, I find some of them empty on google drive.

for instant 1GB.mp4 file transferred to google drive empty with zero size.

I'm using this command for push: drive push -no-clobber -no-prompt /path

Please reply me if I missing something.

Also when i push files to drive, i get this output while the pushing process

trying again in 1.748551918s
trying again in 2.561107794s
trying again in 2.01595369s
trying again in 2.13450818s
trying again in 2.138121141s
trying again in 2.088392877s
trying again in 2.820762315s
trying again in 2.981016122s
trying again in 4.235062082s
trying again in 4.015902713s
trying again in 4.480348395s

is that normal ?

Thank You

**Comment by odeke-em** [comment]
Hello again @bopbo.

So firstly `drive version`?
Secondly, would you mind listing the permissions of those files? Also the `trying again in X.Y* seconds`, is from an exponential back off retry mechanism -- that's expected only if you have `--verbose` defined.
Looks like a no proper file access, but that's just a thought.

**Comment by bopbo** [comment]
# drive version
drive version: 0.2.9
Commit Hash: <CURRENT_COMMIT>
Go Version: <GO_VERSION>
OS: <OS_INFO>
BuildTime: 

Files permission: 755, 777, 644

I don't think this's about file permission.

I will give a real sample:

- We have 100 mp4 files named (001.mp4, 002.mp4, 003.mp4 to 100.mp4)
- All files in one local folder ex.(/test/one) with the same permission

When pushing these files to drive, we get 8 files with zero bytes in google drive.
These files are (014, 015, 090, 055, 074, ...)

We deleted the whole folder from google drive and pushing again.
we get the previous missing files working and another random 11 files have zero bytes. (010, 019, 043, 021, 063, ...)

which mean there's nothing wrong with the file itself.

**Comment by bopbo** [comment]
Also the zero bytes file get word icon instead of video icon as shown in the screenshot.

![screen shot](https://cloud.githubusercontent.com/assets/6955929/9701250/40618150-5429-11e5-9fa8-79d07b3560e4.png)

**Comment by odeke-em** [comment]
Oh okay I see, that is really odd. I use drive to upload lots of video files but I haven't yet encountered this issue. How about https://github.com/odeke-em/drive/issues/345? Btw would you mind running
```shell
$ drive stat 003.mp4
```

**Comment by odeke-em** [comment]
Ping!

**Comment by odeke-em** [comment]
Can't reproduce and a retry will be invoked if any errors occur. Please get the latest from master. Closing this issue. Please reopen if it persists! Thank you for reporting it.

**Comment by dman61** [comment]
I was able to duplicate this issue. If I do:

/usr/bin/drive push -no-clobber -verbose -no-prompt myfile

The file uploaded will be 0 bytes. Drive list also shows them as nil. 
It appears to have something to do with the -no-clobber flag. If I remove that flag then the files upload fine.

Drive version 0.3.1

**Comment by raintonr** [comment]
I've seen this happen too. I'm syncing image files and sometimes see zero size versions in drive after a push.

'drive stat' shows the Google version is zero bytes even though the local obviously isn't.

I was able to re-push this file, but the point is it shouldn't have ever ended up on drive with the zero size.

**Comment by osuka** [comment]
This just happened to me with version 0.3.3/go 1.5.1 on a Mac. I uploaded around 40GB of data, with multiple files.
Two kinds of errors:
1) The file is uploaded as zero bytes
2) The file is uploaded twice, one with correct size, one with zero bytes

In case of (2), I get errors doing a pull/diff/push afterwards, it stops complaining about a name clash (that it itself generated, I'm on non-case sensitive filesystem).

I've been trying to fix this by going to drive web and delete the zero byte files (for case 2) and then hopefully be able to try re-push for case 1.

The problem is that drive drive only reports errors for a folder and then it stops. I fix those manually and have to retry pull, wait, and get another error for a different folder (and a stop).

Any pointers? As it stands I can't really use it.

**Comment by odeke-em** [comment]
Hey @osuka what do your file permissions look like for those files?

**Comment by odeke-em** [comment]
Re-opened this issue.

**Comment by osuka** [comment]
Nothing special:
```
-rwxr-xr-x@ 1 osuka  staff  975696 13 Oct  2013 folder_folder_folder/chap1_66.svg
```

These files were moved from a Dropbox folder as I was attempting migration to Google Drive (luckily I made a copy before). Some other files have different permissions (700 or 600) but this doesn't seem to make a difference, there's many non-conflicting files with the same permissions.

**Comment by odeke-em** [comment]
@osuka, if this file isn't confidential, would you mind mailing it to me, am currently infront of my computer and trying to figure this out. thanks. BTW welcome to drive!

**Comment by odeke-em** [comment]
@osuka also please pass in flag `--verbose` during a push or pull so that you can see the actual fails.

**Comment by odeke-em** [comment]
Like this
```shell
$ drive push --verbose [other_args....]
```

**Comment by osuka** [comment]
Just emailed that file and the --verbose output of an attempted pull (although the output didn't seem to change with the --verbose flag).

**Comment by odeke-em** [comment]
Thanks @osuka, if you don't mind, I'll report the result here
<img width="574" alt="screen shot 2015-12-01 at 10 13 00 am" src="https://cloud.githubusercontent.com/assets/4898263/11508093/316cc734-9814-11e5-8bdb-3202f5e446fb.png">
but am on some super fast network so it wouldn't reproduce this issue easily.

So a quick question, did you try pushing with more than one instance of drive?

**Comment by osuka** [comment]
I don't think it's to do with the size, type or contents of the file, I have (from what I can gather) at least about 100 files with this issue, and they are completely unrelated.

My network speed here at work is also pretty fast (around 100Mbps).

I only had one instance of drive running, although there's two other devices with the official google drive client too (iphone, android). Even though, the only one writing (or doing any kind of modification) was the laptop where the errors originated.

I can try removing all files and re-uploading, maybe with verbose activated (not sure if that would make a difference).

**Comment by odeke-em** [comment]
@osuka as mentioned in my email reply to you, for now please follow the step in the warning
`--fix-clashes` or try removing them manually. My apologies, in the future when am free, I'll create a command to auto remove duplicates https://github.com/odeke-em/drive/issues/510.

**Comment by odeke-em** [comment]
Size? Hmm, I occasionally upload files greater than 20MB so the small size shouldn't trip things out. But good point right there. Sure please try that and lets see.

**Comment by osuka** [comment]
Hi, --fix-clashes would work for "case 2" above, although with an annoying rename of files that can break my file references (eg an SVG that includes a reference to another one by name). I could do that, but for "case 1" above, where files have been uploaded just one but with size 0, if I try pull it tries to remove my local files, replacing them with the zero-byte version.

I'll retry this and see if I can figure out what's happening, and report tomorrow. Thanks for your quick replies.

**Comment by odeke-em** [comment]
Yeah that's why I created that issue for a feature request to auto remove dups.

**Comment by osuka** [comment]
retried this:
- A 4.5GB folder
- drive push (verbose doesn't seem to log anything at all)
- drive pull: (without any other operation in between). Lists a files it wants to delete/modify, which I can't understand:
```
Deletion count 135 dest: 567.87MB
Modification count 127 dest: 57.85MB
```
From what I see, files it wants to delete are files that silently failed to upload, and files it wants to modify are files that were uploaded as size 0. I've taken a look at the code trying to figure out where the problem could be, but I lack skills in go.

**Comment by jpambrun** [comment]
I have the same issue with ver 0.3.3. I had to keep doing "drive push folder/" in order to reduce the "modification count" to zero. The files that needed to be re-uploaded had a '-' file size on Google Drive.

I ran it about 10 times before everything was up-to-date. 
The last few attempts :
- 6 files were correctly updated out of 12 remaining zero length files;
- 1 files were correctly updated out of 6 remaining zero length files;
- 3 files were correctly updated out of 5 remaining zero length files;
- 3 files were correctly updated out of 5 remaining zero length files;
- 2 files were correctly updated out of 2 remaining zero length files.

I had another instance pushing files from another computer at the same time except for the initial upload that left dozens of zero files a few days ago.

**Comment by dimatura** [comment]
Exact same problem (zero length files, showing up as '-' in the drive page) here with v0.3.4, uploading a 4.7GB with 2136 files, of which 1816 were uploaded the first time around. Platform is Ubuntu 14.04 x86_64.

**Comment by jp-uom** [comment]
I also have this problem with 0.3.9.1 on Ubuntu 16.
